### ISSUE TITLE
4.9.5， error `Decorators are not valid here. `

### ISSUE THREAD

**Issue by wangyang875** [microsoft/TypeScript#55742]
### 🔎 Search Terms

decorator 

### 🕗 Version & Regression Information

- This is a error
- This changed between versions 4.9.5 and latter version

### 💻 Code

```ts
// Your code here
export default () => {
  return class C {
    constructor(@Foo() param: any) { }
  }
}
function Foo(...args: any[]): any {}
```

### 🙁 Actual behavior

@Foo will report error `Decorators are not valid here.`

### 🙂 Expected behavior

no error should be reported

**Comment by MartinJohns** [comment]
> This changed between versions 4.9.5 and latter version

I can not confirm this. There's no difference between 4.8.4, 4.9.5 and 5.0.4.
